tests/test_opds.py

```diff
@@ -1348,6 +1349,62 @@ def test_acquisition_links(self):
             hold.license_pool, None, hold, None, feed, hold.license_pool.identifier)
         eq_([], hold_links)
 
+    def test_acquisition_links_multiple_links(self):
+        annotator = LibraryLoanAndHoldAnnotator(None, None, self._default_library, test_mode=True)
+        feed = AcquisitionFeed(
+            self._db, "test", "url", [], annotator
+        )
+
+        # This book has two delivery mechanisms
+        work = self._work(with_license_pool=True)
+        [pool] = work.license_pools
+        [mech1] = pool.delivery_mechanisms
+        mech2 = pool.set_delivery_mechanism(
+            Representation.PDF_MEDIA_TYPE, DeliveryMechanism.NO_DRM,
+            RightsStatus.IN_COPYRIGHT, None
+        )
+
+        # The vendor API for LicensePools of this type requires that a
+        # delivery mechanism be chosen at the point of borrowing.
+        class MockAPI(object):
+            SET_DELIVERY_MECHANISM_AT = BaseCirculationAPI.BORROW_STEP
+
+        # This means that two different acquisition links will be
+        # generated -- one for each delivery mechanism.
+        links = annotator.acquisition_links(
+            pool, None, None, None, feed, pool.identifier,
+            mock_api=MockAPI()
+        )
+        eq_(2, len(links))
+        indirects = []
+        for link in links:
+            # Both links should have the same subtags.
+            [availability, copies, holds, indirect] = sorted(
+                link, key=lambda x: x.tag
+            )
+            assert availability.tag.endswith('availability')
+            assert copies.tag.endswith('copies')
+            assert holds.tag.endswith('holds')
+            assert indirect.tag.endswith('indirectAcquisition')
```

Comment:
This part tests the bug fix. Previously only the second tag would have the availability, copies, or holds tags.